packages/lib/src/components/internal/Address/Address.tsx

```diff
@@ -50,10 +50,11 @@ export default function Address(props: AddressProps) {
     });
 
     const setSearchData = selectedAddress => {
+        const updatedSelectedAddress = props.onSelectedAddress ? props.onSelectedAddress(selectedAddress) : selectedAddress;
```

Comment:
We might need await here @kunalkp  maybe do a test with async function

Reply:
Yes. Fixed!